src/jupyterhub_config/custom_docker_spawner.py

```diff
@@ -131,3 +131,39 @@ def _configure_notebook_dir(self, username: str, user_dir: Path):
         else:
             self.log.info(f'Non-admin user detected: {username}. Setting up user-specific workspace.')
             self.notebook_dir = str(user_dir)
+
+    def _is_rw_minio_user(self):
+        """
+        Check if the user is a read/write MinIO user.
+
+        Admin users and users in the minio_rw group are considered read/write MinIO users.
+        """
+        group_names = [group.name for group in self.user.groups]
+        return self.user.admin or self.RW_MINIO_GROUP in group_names
+
+    def _ensure_user_volume(self):
+        """
+        Ensure the user's volume is correctly mounted in the container.
+        """
+
+        user_home_dir = Path(os.environ['JUPYTERHUB_USER_HOME'])
+        mount_base_dir = Path(os.environ['JUPYTERHUB_MOUNT_BASE_DIR'])
+        hub_secrets_dir = Path(os.environ['JUPYTERHUB_SECRETS_DIR'])
```

Comment:
Remind me what's in the secrets dir, I've already forgotten

Reply:
it holds cookie sessions and user DB.

Reply:
:+1: